File: Python/Aulas/tap/treino_2/1968.py
def crivo(a, n, k):
    tabela = [0]*(n + 1)
    kDivisores = []
    
    R = int(n ** 0.5)
    for i in range(1, R + 1):
        tabela[i * i] += 1
        
        for j in range(i*i + i, n + 1, i):
            tabela[j] += 2
            
        
    
    for i in range(a, n + 1):
        if tabela[i] == k:
            kDivisores.append(i)

    return kDivisores

def main():
    a, b, k = [int(x) for x in input().split()]

    lideres = len(crivo(a, b, k))

    resp = lideres ** (((b - a) + 1) - lideres)
    print(resp)

main()

File: Python/Aulas/tap/treino_2/1968.cpp
#include <iostream>
#include <math.h>

using namespace std;
typedef unsigned long long ll;

ll tabela[10000001];
ll a, b, k, lideres, pessoas, resp;

ll crivo(ll a, ll n, ll k)
{
	ll i = 1;
	while (i * i <= n)
	{
		tabela[i * i] += 1;
		
		for(ll j = i * i + i; j <= n; j += i)
		{
			tabela[j] += 2;
		}
		i++;
	}
	
	ll lid = 0;
	for (ll i = a; i <= n; i++)
	{
		if (tabela[i] == k) 
		{
			lid = (ll)(lid + 1);
		}
	}
	
	return lid;	
}

ll pont(ll base, ll exp)
{
	ll r = 1;
	while (exp > 0)
	{
		r = (ll)(r * base) ;
		r %= 1000000007;
		exp -= 1;
	}
	return r;
}

int main()
{
	cin >> a >> b >> k;
	lideres = crivo(a, b, k);
	pessoas = ((b - a) + 1) - lideres;

	cout << pont(lideres, pessoas) << endl;
}
